src/Elm/Parser/Declarations.elm

```diff
@@ -592,71 +611,66 @@ customTypeDefinitionAfterDocumentationAfterTypePrefix =
 
 typeOrTypeAliasDefinitionWithoutDocumentation : Parser (WithComments (Node Declaration.Declaration))
 typeOrTypeAliasDefinitionWithoutDocumentation =
-    ParserFast.mapWithStartPosition
-        (\start result ->
-            { comments = result.comments
-            , syntax =
-                Node { start = start, end = result.end }
-                    result.declaration
-            }
-        )
-        (ParserFast.map2
-            (\commentsAfterType afterStart ->
-                let
-                    allComments : Comments
-                    allComments =
-                        commentsAfterType |> Rope.prependTo afterStart.comments
-                in
-                case afterStart.syntax of
-                    TypeDeclarationWithoutDocumentation typeDeclarationAfterDocumentation ->
-                        let
-                            end : Location
-                            end =
-                                case typeDeclarationAfterDocumentation.tailVariantsReverse of
-                                    (Node range _) :: _ ->
-                                        range.end
-
-                                    [] ->
-                                        let
-                                            (Node headVariantRange _) =
-                                                typeDeclarationAfterDocumentation.headVariant
-                                        in
-                                        headVariantRange.end
-                        in
-                        { comments = allComments
-                        , declaration =
-                            Declaration.CustomTypeDeclaration
+    ParserFast.map2WithStartLocation
```

Comment:
We do some computations here to get the `end` position. Would it be faster to use `map2WithRange`?

Reply:
That would mean unnecessary computation and application for all other declarations. A case-of is pretty cheap anyway and has the added benefit the we could easily get rid of the backtracking for the last variant.